internal/init_test.go

```diff
@@ -0,0 +1,15 @@
+package internal
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBundleInitErrorOnUnknownFields(t *testing.T) {
+	t.Log(GetEnvOrSkipTest(t, "CLOUD_ENV"))
```

Comment:
For now we need to skip becuase auth credentials are required. See failed run: https://github.com/databricks/cli/actions/runs/6157685546/job/16708920147

I suspect this is happening because of this: https://github.com/databricks/cli/blob/96d807fb858ef0a413497ea32c86d117f4f7d91d/libs/template/helpers.go#L33, but I have not verified this yet.

Reply:
@shreyas-goenka can you move ` w := root.WorkspaceClient(ctx) ` inside the helpers where it's used, like `smallest_node_type`? Then it will be lazily initialised.

Reply:
Meh, I think it's fine if we always require a workspace client to be configured to run bundle init. In fact, it should really be initialized even earlier perhaps.

Having a host parameter in a bundle template should be pretty common.

Reply:
`w := root.WorkspaceClient(ctx)` queries the context for the workspace client, the initialization happens somewhere up in the call stack. 

I created an internal followup ticket to track this. It's not just an issue for e2e tests, bundle init itself returns the same error.I agree that this dependency should be a lazy evaluation, and the error message needs to be clearer.

Reply:
@pietern The error message currently is not great though, and that could use some work. Also allows tests on PR which might be nice if it's an easy enough change to make.